### PR TITLE
[PRIM][IR]support more vjp code gen

### PR DESCRIPTION
**Pull request by Charles-hit**
### PR types

New features 
### PR changes

Others
### Description

Pcard-66975
support more vjp code gen in new ir.

### CODE REVIEW

File: test/prim/new_ir_prim/test_vjp_prim.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def test_divide_grad_prim_case1(self):
         ]
         for idx, op in enumerate(newir_program.block().ops):
             self.assertEqual(op.name(), all_op_names[idx])
+        paddle.fluid.core._set_prim_backward_enabled(False)
```
Comment (cxxly) [inline test/prim/new_ir_prim/test_vjp_prim.py]: paddle.framework.coer.xxx

File: test/prim/new_ir_prim/test_vjp_prim.py [hunk 2]
```diff
@@ -145,6 +146,7 @@ def test_sum_grad_prim(self):
...
```
Comment (cxxly) [inline test/prim/new_ir_prim/test_vjp_prim.py]: paddle.framework.core.xxx

File: paddle/fluid/operators/generator/tests_utils.py [hunk 3]
```diff
@@ -89,16 +89,14 @@ def is_tensor_list(s):
 def exist_mutable_attribute(attrs):
     for attr in attrs:
         if (
-            attr['typename'] in ['Scalar', 'IntArray']
-            and attr['support_tensor'] is True
-        ):
+            is_scalar(attr['typename']) or is_intarray(attr['typename'])
+        ) and attr['support_tensor'] is True:
```
Comment (cxxly) [inline paddle/fluid/operators/generator/tests_utils.py]: attr['support_tensor'] is True -> attr['support_tensor']

File: paddle/fluid/operators/generator/tests_utils.py [hunk 4]
```diff
@@ -89,16 +89,14 @@ def is_tensor_list(s):
 def exist_mutable_attribute(attrs):
```
Comment (cxxly) [inline paddle/fluid/operators/generator/tests_utils.py]: exist_mutable_attr 保持命名一执性，缩写全用缩写，全称全用全称，避免混用

File: paddle/fluid/operators/generator/tests_utils.py [hunk 5]
```diff
@@ -89,16 +89,14 @@ def is_tensor_list(s):
...
+        return False
 
 
 def is_mutable_attribute(attr):
     return (
-        attr['typename'] in ['Scalar', 'IntArray']
-        and attr['support_tensor'] is True
-    )
+        is_scalar(attr['typename']) or is_intarray(attr['typename'])
+    ) and attr['support_tensor'] is True
```
Comment (cxxly) [inline paddle/fluid/operators/generator/tests_utils.py]: attr.get('support_tensor', False)

**Review by cyber-pioneer — APPROVED** [review]
LGTM

**Review by heavyrain-lzy — APPROVED** [review]
LGTM

File: paddle/fluid/operators/generator/tests_utils.py (hunk 3 above)
Comment (Charles-hit) [reply paddle/fluid/operators/generator/tests_utils.py]: done

File: paddle/fluid/operators/generator/tests_utils.py (hunk 4 above)
Comment (Charles-hit) [reply paddle/fluid/operators/generator/tests_utils.py]: done

File: paddle/fluid/operators/generator/tests_utils.py (hunk 5 above)
Comment (Charles-hit) [reply paddle/fluid/operators/generator/tests_utils.py]: done

**Pull request merged**